src/services/download-service.ts

```diff
@@ -70,20 +73,47 @@ class DownloadService {
       const downloadUrl = response.url;
 
       const isContentTypeSupported = this.player.isImage() || this.isContentTypeSupported(response);
-      const fileName = this.player.isImage() ? this.player.sources?.metadata?.name || 'Image' : this.getFilename(response);
+      const fileName = this.getFilename(response);
 
       if (isContentTypeSupported && fileName) {
         return {
           downloadUrl,
           fileName
         };
       }
-    } catch (e: any) {}
+    } catch (e: any) {
+      // in case HEAD request failed for raw service (image use-case), retry to get the image download metadata using thumbnail service
+      if (this.player.isImage()) {
+        return await this.getImageDownloadMetadata();
+      }
+    }
+
+    return null;
+  }
 
+  private async getImageDownloadMetadata(): Promise<DownloadMetadata> {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const imageSource = this.player.config.sources.image;
```

Comment:
is it possible that sources will not be available like in line 25

Reply:
actually at this point sources must be available, the first time it happens is after loadMedia.